pkg/controllers/workgenerator/controller_test.go

```diff
@@ -971,10 +970,26 @@ func TestSetBindingStatus(t *testing.T) {
 	}()
 	for name, tt := range tests {
 		t.Run(name, func(t *testing.T) {
-			maxFailedResourcePlacementLimit = tt.maxFailedResourcePlacementLimit
+			if tt.maxFailedResourcePlacementLimit != nil {
+				maxFailedResourcePlacementLimit = *tt.maxFailedResourcePlacementLimit
+			}
 			binding := &fleetv1beta1.ClusterResourceBinding{}
 			setBindingStatus(tt.works, binding)
 			got := binding.Status.FailedPlacements
+			// setBindingStatus is using map to populate the failedResourcePlacement.
+			// There is no default order in traversing the map.
+			// When the result of failedResourcePlacement exceeds the limit, the result will be truncated and cannot be
+			// guaranteed.
```

Comment:
I was aware of that but thought it was not super random. Now that we don't test the limit anymore, can we just remove the limit from the test?

Reply:
we're testing limit in line 821